### ISSUE TITLE
Release date keeps getting updated even if you're updating the content of the release notes

### ISSUE THREAD

**Issue by richmahn** [go-gitea/gitea#12341]
- Gitea version (or commit ref): 1.12.2
- Git version: 2.14.5
- Operating system: MacOS
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

When I make a release, the created unix time gets set to the time I make the release, no the commit, which is right.

Yet if I now update my release by editing its content, its date changes.

This is a problem if I update an old release, such as tell users not to use, it becomes the release at the top of the release page!

Related to this change: https://github.com/go-gitea/gitea/pull/11218

Basically you should only change the release created_unix if it is making an actual tag (going from Draft to pre-release or release), do not touch it if updating a release or pre-release.

...

## Screenshots
On try.gitea.io, making two releases v1, then v2, then edit v1. It now is at the top.

![Screen Shot 2020-07-27 at 3 39 54 PM](https://user-images.githubusercontent.com/2839925/88584452-a213ed80-d01f-11ea-8d18-6f2744941feb.png)
![Screen Shot 2020-07-27 at 3 40 14 PM](https://user-images.githubusercontent.com/2839925/88584457-a2ac8400-d01f-11ea-9174-176a95d60822.png)
![Screen Shot 2020-07-27 at 3 40 32 PM](https://user-images.githubusercontent.com/2839925/88584460-a3451a80-d01f-11ea-8fc1-8272b24e51af.png)

**Comment by richmahn** [comment]
I'll probably end up working on this anyway. Will make a fix.
